### ISSUE TITLE
Need Gtest and Some Pip Installs for Docker Images

### ISSUE THREAD

**Issue by herter4171** [idaholab/moose#15981]
## Reason

I was told that Gtest was needed for tinkering with unit tests, so that needs to be added.  I also have some `pip3` installs needed for certain clustering behavior for BLACKBEAR/GRIZZLY and BISON.

## Design

In directory `docker_ci`, I will add 
* `gtest` to `yum_installs.sh`
* `libgtest-dev` to `apt_installs.sh`
* `pip3` packages to `install_packages.sh` 
    * `python-consul`
    * `python-nomad`

Basically, for multi-node container jobs, we use the Consul API to get the head node IP, and from there, we use Nomad's API to get node IPs to dynamically build a machine file for `mpirun` to point to at launch.

## Impact

This will ensure the images have `gtest` installed, and the `pip3` installs won't impact anything.
